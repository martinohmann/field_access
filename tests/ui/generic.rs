use field_access::FieldAccess;

#[derive(FieldAccess)]
pub struct Foo<T> {
    a: T,
}

fn main() {}
