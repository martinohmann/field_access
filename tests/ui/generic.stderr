error[E0310]: the parameter type `T` may not live long enough
 --> tests/ui/generic.rs:3:10
  |
3 | #[derive(FieldAccess)]
  |          ^^^^^^^^^^^
  |          |
  |          the parameter type `T` must be valid for the static lifetime...
  |          ...so that the type `Foo<T>` will meet its required lifetime bounds...
  |
note: ...that is required by this bound
 --> src/lib.rs
  |
  | pub trait AnyFieldAccess: Any {
  |                           ^^^
  = note: this error originates in the derive macro `FieldAccess` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider adding an explicit lifetime bound
  |
4 | pub struct Foo<T: 'static> {
  |                 +++++++++

error[E0310]: the parameter type `T` may not live long enough
 --> tests/ui/generic.rs:3:10
  |
3 | #[derive(FieldAccess)]
  |          ^^^^^^^^^^^
  |          |
  |          the parameter type `T` must be valid for the static lifetime...
  |          ...so that the type `T` will meet its required lifetime bounds...
  |
note: ...that is required by this bound
 --> src/lib.rs
  |
  | pub trait AnyFieldAccess: Any {
  |                           ^^^
  = note: this error originates in the derive macro `FieldAccess` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider adding an explicit lifetime bound
  |
4 | pub struct Foo<T: 'static> {
  |                 +++++++++
